src/test/java/com/google/sps/ProfileBuilderServletTest.java

```diff
@@ -0,0 +1,196 @@
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.gson.Gson;
+import com.google.sps.data.User;
+import com.google.sps.data.PersistentUserRepository;
+import com.google.sps.servlets.ProfileBuilderServlet;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public class ProfileBuilderServletTest {
+    
+  @Mock
+  private HttpServletRequest request;
+
+  @Mock
+  private HttpServletResponse response;
+
+
+  private ProfileBuilderServlet userServlet;
+
+  private LocalServiceTestHelper localHelper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+    
+    
+  @Before
+  public void setUp() throws Exception {
+    userServlet = new ProfileBuilderServlet(); 
+    request = Mockito.mock(HttpServletRequest.class);       
+    response = Mockito.mock(HttpServletResponse.class);
+
+    localHelper.setUp();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    localHelper.tearDown();
+  }
+    
+  public void addMentorProfile() {
+    String userType = "Mentor";
+    when(request.getParameter("user-Type")).thenReturn(userType);
+    String mentorName = "Jeff";
+    when(request.getParameter("profName-input")).thenReturn(mentorName);
+    String company = "Google";
+    when(request.getParameter("company-input")).thenReturn(company);
+    String specialty = "Security";
+    when(request.getParameter("specialty-input")).thenReturn(specialty);
+  }
+
+  public void addMenteeProfile() {
+    String userType = "Mentee";
+    when(request.getParameter("user-Type")).thenReturn(userType);
+    String menteeName = "Bobby";
+    when(request.getParameter("name-input")).thenReturn(menteeName);
+    String school = "University of California Berkeley";
+    when(request.getParameter("school-input")).thenReturn(school);
+    String major = "Computer Science";
+    when(request.getParameter("major-input")).thenReturn(major);
+  }
+
+  @Test
+  public void doPostRedirectsUrl() throws IOException, ServletException {
+    addMentorProfile();
+    userServlet.doPost(request, response);
+
+    verify(response, times(1)).sendRedirect("/logged_in_homepage.html");
+   }
+  
+  @Test
+  public void addMentorProfileThenCheckSize() throws IOException, ServletException {
+    DatastoreService dataService = DatastoreServiceFactory.getDatastoreService();
+    addMentorProfile();
+    userServlet.doPost(request, response);
```

Comment:
Could you add additional tests to ensure your servlet works as expected with multiple or zero mentor/mentees?

Reply:
Made test that adds Mentee, post. Then add Mentor, and post

Reply:
Done